connectors/vault/vault_connector.go

```diff
@@ -5,52 +5,56 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"log"
 	"net"
 	"strconv"
-	"strings"
 
+	. "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	vaultbl "github.com/ibm/the-mesh-for-data/connectors/vault/vaultconn-bl"
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 	"google.golang.org/grpc"
 )
 
-var vault VaultConnection
-
 const defaultTimeout = "180"
 const defaultPort = "50083" //synched with vault_connector.yaml
 
 type server struct {
 	pb.UnimplementedDataCredentialServiceServer
+	vaultConnector *vaultbl.Server
 }
 
-func (s *server) GetCredentialsInfo(ctx context.Context, in *pb.DatasetCredentialsRequest) (*pb.DatasetCredentials, error) {
-	log.Println("Vault connector: GetCredentialsInfo")
-	log.Printf("Vault connector: GetCredentialsInfo, in = %v\n", in)
+var vault VaultConnection
 
-	// secretaddr := url.QueryEscape(in.DatasetId)
-	secretaddr := strings.ReplaceAll(in.DatasetId, " ", "")
+func (s *server) GetCredentialsInfo(ctx context.Context, in *pb.DatasetCredentialsRequest) (*pb.DatasetCredentials, error) {
+	log.Println("Received ApplicationContext")
+	log.Println(in)
 
-	readCredentials, err := vault.GetFromVault(secretaddr)
+	vaultPathKey := GetEnv(VaultPathKey)
+	eval, err := s.vaultConnector.GetCredentialsInfo(in, vault, vaultPathKey)
 	if err != nil {
 		log.Printf("Error in vaultConnector, got error from the vault: %v\n", err.Error())
-		return nil, fmt.Errorf("error in retrieving the secret from vault(key = %s): %v", secretaddr, err)
+		return nil, fmt.Errorf("error in retrieving the secret from vault: %v", err)
 	}
-	log.Println("Read credentials from vault: " + readCredentials)
+	jsonOutput, _ := json.MarshalIndent(eval, "", "\t")
+	log.Println("Received evaluation : " + string(jsonOutput))
+	log.Println("err:", err)
+
+	return eval, err
 
-	return &pb.DatasetCredentials{DatasetId: in.DatasetId, Credentials: readCredentials}, nil
 }
 
 func main() {
-	vaultAddress := getEnv(VaultAddressKey)
-	timeOutInSecs := getEnvWithDefault(VaultTimeoutKey, defaultTimeout)
+	vaultAddress := GetEnv(VaultAddressKey)
+	timeOutInSecs := GetEnvWithDefault(VaultTimeoutKey, defaultTimeout)
 	timeOutSecs, err := strconv.Atoi(timeOutInSecs)
-	port := getEnvWithDefault(VaultConnectorPortKey, defaultPort)
+	port := GetEnvWithDefault(VaultConnectorPortKey, defaultPort)
 
 	log.Printf("Vault address env variable in %s: %s\n", VaultAddressKey, vaultAddress)
 	log.Printf("VaultConnectorPort env variable in %s: %s\n", VaultConnectorPortKey, port)
 	log.Printf("TimeOut used %d\n", timeOutSecs)
-	log.Printf("Secret Token env variable in %s: %s\n", VaultSecretKey, getEnv(VaultSecretKey))
+	log.Printf("Secret Token env variable in %s: %s\n", VaultSecretKey, GetEnv(VaultSecretKey))
```

Comment:
Ok. You really have to stop logging credentials. Not only is the VaultSecretKey logged explicitly here but also in the `GetEnv` method.

Reply:
fixed. Thanks.